File: jja6312/2월24일/구명보트.java
import java.util.Arrays;

public class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people); // 몸무게를 오름차순으로 정렬
        int i = 0, j = people.length - 1; // 가장 가벼운 사람과 가장 무거운 사람의 인덱스
        int answer = 0; // 사용된 구명보트의 수

        while (i <= j) {
            if (people[i] + people[j] <= limit) {
                // 두 사람을 함께 보낼 수 있음
                i++;
            }
            j--;
            answer++;
        }

        return answer;
    }
}

File: zuzubibi/2월/2월 24일/구명보트.py
def solution(people, limit):
    people.sort()
    cnt = 0
    front, end = 0, len(people)-1

    while front <= end:
        if people[front] + people[end] <= limit:
            front += 1
        
        end -= 1
        cnt += 1

    return cnt

print(solution([70, 50, 80, 50], 100)) # 3
print(solution([70, 80, 50], 100)) # 3